### DISCUSSION TITLE
Revamp Searching for Metrics

### DISCUSSION THREAD

**Discussion by Sebastiaan127001** [ICTU/quality-time#6255] [Ideas]
I suggest we completely change the way the user can search for metrics.
At this moment, Quality-time itself only offers the following ways to search/filter, apart from searching text with your browser.

- Metrics requiring action.
- Metrics with a certain label.
- Metrics within a certain subject.

We could add a dynamic metric filtering features by adding a bar on the top of a report.
This is more or less how one can search in Jira for issues.

The "metrics requiring action" could be moved to this metrics search feature.
We could add the following to this.
- Metrics having an issue linked to it.
- Metrics using source GitLab / Jira / etc. see https://github.com/ICTU/quality-time/issues/3331

It could work like so: https://www.kindacode.com/article/how-to-create-a-filter-search-list-in-react/

What are your thoughts on this?

**Comment by fniessink** [comment]
Possible inspirations:
- GitLab pipeline filter/search bar 
- Jira basic search

This search bar would realise #3331 and #6256 (which can be closed when we have an issue for the search bar).
